src/App.tsx

```diff
@@ -15,36 +15,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+export const App = () => {
+  const [sortType, setSortType] = useState('default');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const getSortedGoods = () => {
+    const sortedGoods = [...goodsFromServer];
+
+    if (sortType === 'alphabet') {
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+    } else if (sortType === 'length') {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
+
+    return sortedGoods;
+  };
+
+  const handleSortAlphabetically = () => {
+    setSortType('alphabet');
+  };
+
+  const handleSortByLength = () => {
+    setSortType('length');
+  };
+
+  const handleReverse = () => {
+    setIsReversed(prevState => !prevState);
+  };
+
+  const handleReset = () => {
+    setSortType('default');
+    setIsReversed(false);
+  };
+
+  const isActive = sortType !== 'default' || isReversed;
+  const sortedGoods = getSortedGoods();
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${sortType === 'alphabet' ? '' : 'is-light'}`}
+          onClick={handleSortAlphabetically}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-success ${sortType === 'length' ? '' : 'is-light'}`}
+          onClick={handleSortByLength}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={`button is-warning ${isReversed ? '' : 'is-light'}`}
+          onClick={handleReverse}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {isActive && (
+          <button
+            type="button"
+            className="button is-danger"
+            onClick={handleReset}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {sortedGoods.map(good => (
+          <li key={good} data-cy="Good">
```

Comment:
[CODE KNOWLEDGE] - When you rendering a list, don't forget to add `key` prop; It's good that the `key` prop is provided for each list item, but using the good's name might cause issues if the list contains duplicate names. Consider using a unique identifier instead.